src/builder.rs

```diff
@@ -598,11 +691,32 @@ impl Builder {
         }
 
         self.outputs
-            .push(OutputInfo::new(ovk, recipient, value, memo));
+            .push(OutputInfo::new(ovk, recipient, value, asset, memo));
 
         Ok(())
     }
 
+    /// Add an instruction to burn a given amount of a specific asset.
```

Comment:
To follow the Rust style guide, and our existing convention:
```suggestion
    /// Adds an instruction to burn a given amount of a specific asset.
```